### PR TITLE
Ensuring all receivers are tested

### PR DESCRIPTION
**Pull request by MovieStoreGuy**
**Description:**
Adding all receivers to be tested as part of the unit test run on merge.

**Link to tracking Issue:**
Currently no issue but more of a cleanup

**Testing:**
Only tests were updated

**Documentation:** 
N/A

### CODE REVIEW

File: internal/components/receivers_test.go [hunk 1]
```diff
@@ -79,21 +162,82 @@ func TestDefaultReceivers(t *testing.T) {
...
+		},
+		{
+			receiver:     "windowsperfcounters",
+			skipLifecyle: runtime.GOOS != "windows", // Only works on windows based machines
+		},
 		{
 			receiver: "zipkin",
 		},
+		{
+			receiver:     "zookeeper",
```
Comment (MovieStoreGuy) [inline internal/components/receivers_test.go:202]: This requires a follow up issue to make the shutdown work without issue with the default configuration.

**Review by MovieStoreGuy — COMMENTED** [review]
Few things to double check but this is all the current receivers being added to the testing suite.

File: internal/components/receivers_test.go [hunk 2]
```diff
@@ -47,16 +56,86 @@ func TestDefaultReceivers(t *testing.T) {
...
+		{
+			receiver:     "awsecscontainermetrics",
+			skipLifecyle: true, // Requires container metaendpoint to be running
+		},
+		{
+			receiver:     "awsxray",
+			skipLifecyle: true, // Requires AWS endpoint to check identity to run
+		},
+		{
+			receiver: "carbon",
```
[the remark this answers is not in the document]
Comment (MovieStoreGuy) [reply internal/components/receivers_test.go:68]: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6978

Issue to follow up here:

File: internal/components/receivers_test.go (hunk 1 above)
Comment (MovieStoreGuy) [reply internal/components/receivers_test.go:203]: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6979

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**